src/python/WMCore/Services/WMArchive/CMSSWMetrics.py

```diff
@@ -0,0 +1,503 @@
+            "Timing-tstoragefile-writeViaCache-minMsecs": 0,
+            "Timing-tstoragefile-write-numSuccessfulOperations": 7307,
+            "Timing-tstoragefile-seek-numSuccessfulOperations": 12044,
+            "Timing-tstoragefile-readViaCache-totalMsecs": 0,
+            "Timing-tstoragefile-write-maxMsecs": 12.2428,
+            "Timing-file-read-totalMegabytes": 5.43017,
+            "Timing-tstoragefile-stat-numSuccessfulOperations": 697,
+            "Timing-root-construct-maxMsecs": 0.000261,
+            "Timing-tstoragefile-construct-minMsecs": 36.3145,
+            "Timing-root-writev-maxMsecs": 0,
+            "Timing-tstoragefile-readActual-totalMsecs": 18677.2,
+            "Timing-tstoragefile-construct-totalMsecs": 11419,
+            "Timing-root-writev-totalMsecs": 0,
+            "Timing-file-construct-maxMsecs": 0.000291,
+            "Timing-file-prefetch-minMsecs": 0,
+            "Timing-root-stagein-numOperations": 1,
+            "Timing-file-close-minMsecs": 0.190476,
+            "Timing-file-construct-numOperations": 2,
+            "Timing-file-construct-numSuccessfulOperations": 2,
+            "Timing-root-stagein-minMsecs": 4.10695,
+            "Timing-root-write-totalMegabytes": 0,
+            "Timing-root-read-numSuccessfulOperations": 57,
+            "Timing-tstoragefile-writeActual-minMsecs": 0.005877,
+            "Timing-tstoragefile-readViaCache-numOperations": 3,
+            "Timing-tstoragefile-readActual-maxMsecs": 4168.78,
+            "Timing-file-close-numOperations": 1,
+            "Timing-file-position-numOperations": 14759,
+            "Timing-tstoragefile-read-totalMsecs": 2075,
+            "Timing-root-position-maxMsecs": 0.001203,
+            "Timing-root-construct-minMsecs": 0.000142,
+            "Timing-tstoragefile-readViaCache-numSuccessfulOperations": 0,
+            "Timing-file-position-numSuccessfulOperations": 14759,
+            "Timing-tstoragefile-writeViaCache-maxMsecs": 0,
+            "Timing-tstoragefile-writeViaCache-totalMegabytes": 0,
+            "Timing-file-write-numOperations": 7307,
+            "Timing-tstoragefile-flush-totalMsecs": 1847.36,
+            "Timing-root-construct-numOperations": 3,
+            "Timing-tstoragefile-seek-totalMsecs": 18.3395,
+            "Timing-tstoragefile-stat-totalMegabytes": 0,
+            "Timing-root-prefetch-totalMsecs": 0,
+            "Timing-root-prefetch-maxMsecs": 0,
+            "Timing-tstoragefile-writeActual-totalMsecs": 222.926
+        }
+}
+
+def CMSSWMetrics():
+    """
+    Helper function to convert CMSSW_METRICS dict values to their data-types
+    :return: dictionary
+    """
+    rdict = {}
+    for key, mdict in CMSSW_METRICS.items():
+        if key == 'XrdSiteStatistics':
+            rdict[key] = XrdMetrics(mdict)
+            continue
+        ndict = {}
+        for mkey, mval in mdict.items():
+            for ktype in [int, float, str, bool]:
```

Comment:
We can make this algorithm less expensive by changing this line and the if with something like:
```
if isinstance(mval, (int, float, str, bool)):
    ndict[mkey] = type(mval)
```
thus reaching O(n^2) instead of O(n^3).

Reply:
done